### DISCUSSION TITLE
Detecting if Git

### DISCUSSION THREAD

**Discussion by MrHinsh** [devlooped/GitInfo#323] [Q&A]
What is the best way to detect if there is a git repository where the code is running?

https://github.com/nkdAgility/azure-devops-migration-tools/pull/1726

I have a tool that folks often download and run in debug, then submit a support request. I want to detect if git is available, and if so then output some data.

**What is the best way to test if git?**

**Comment by kzu** [comment]
You can check for a non-empty `$(GitRoot)` (meaning there's a git repo)
